services/mac_manager/src/main/java/com/futurewei/alcor/macmanager/service/MacAddressService.java

```diff
@@ -0,0 +1,149 @@
+/*Copyright 2019 The Alcor Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+        you may not use this file except in compliance with the License.
+        You may obtain a copy of the License at
+
+        http://www.apache.org/licenses/LICENSE-2.0
+
+        Unless required by applicable law or agreed to in writing, software
+        distributed under the License is distributed on an "AS IS" BASIS,
+        WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+        See the License for the specific language governing permissions and
+        limitations under the License.
+*/
+package com.futurewei.alcor.macmanager.service;
+
+import com.futurewei.alcor.common.exception.ResourceNotFoundException;
+import com.futurewei.alcor.macmanager.dao.MacPoolRedisRepository;
+import com.futurewei.alcor.macmanager.dao.MacRangeRedisRepository;
+import com.futurewei.alcor.macmanager.dao.MacRedisRepository;
+import com.futurewei.alcor.macmanager.entity.MacAddress;
+import com.futurewei.alcor.macmanager.entity.MacRange;
+import com.futurewei.alcor.macmanager.entity.MacState;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Vector;
+import java.util.concurrent.ThreadLocalRandom;
+
+@Service
+public class MacAddressService {
```

Comment:
For MacAddressService, I would suggest to split into two layers, service layer and service impl layer. 

Service Layer: Show the interface
Service Impl Layer: Show the implementation of the interface. This design allows different implementations later on.

One example is to use the SubManager as an example: 

https://github.com/kevin-zhonghao/alcor/tree/feature/route_manager/services/subnet_manager/src/main/java/com/futurewei/alcor/subnet/service

Reply:
Actually, this is a great idea. I will do it.